### PR TITLE
Updated Time Range Parsing in Report Details 

### PR DESCRIPTION
**Pull request by davidcui-amzn**
*Issue #, if available:*
N/A 
*Description of changes:*
Time range in Report Details for `Visualizations` and `Saved searches` from in-context menu would show as undefined. Updated the parsing so it displays correctly. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: kibana-reports/public/components/main/report_details/report_details.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ export function ReportDetails(props) {
   const parseTimePeriod = (queryUrl: string) => {
     let timeString = queryUrl.substring(
       queryUrl.lastIndexOf('time:'),
-      queryUrl.lastIndexOf('))')
+      queryUrl.indexOf('))')
```
Comment (zhongnansu) [inline kibana-reports/public/components/main/report_details/report_details.tsx:124]: This change doesn't seem to be specific for visualization and saved search, but why it was fine with dashboard before this fix?

Comment (davidcui-amzn) [reply kibana-reports/public/components/main/report_details/report_details.tsx:124]: The url construction for visualization vs saved search vs dashboard is not the same

Comment (zhongnansu) [reply kibana-reports/public/components/main/report_details/report_details.tsx:124]: I get that it's different but the time range part of the url should be the same, the only difference is dashboard#/view vs discover#/edit, as far as I can remember.

Comment (zhongnansu) [reply kibana-reports/public/components/main/report_details/report_details.tsx:124]: This solution of locating a substring with `))` seems kind of hack for me. I'd rather use a more generic solution. Could you try this regex expression to match the time field `/time:\(from:.+,to:.+?\)/`

File: kibana-reports/public/components/main/report_details/report_details.tsx [hunk 2]
```diff
@@ -119,19 +119,18 @@ export function ReportDetails(props) {
...
 
   const parseTimePeriod = (queryUrl: string) => {
-    let timeString = queryUrl.substring(
-      queryUrl.lastIndexOf('time:'),
-      queryUrl.indexOf('))')
-    );
+    let timeStringRegex = /time:\(from:.*,to:.*\)/;
+
+    let timeStringRegexArray = timeStringRegex.exec(queryUrl);
+    let timeString = timeStringRegexArray[0];
```
Comment (joshuali925) [inline kibana-reports/public/components/main/report_details/report_details.tsx:125]: just a note, you can probably do `const [timeString, fromDateString, toDateString] = queryUrl.match(/time:\(from:(.+),to:(.+?)\)/)` might be easier

also I think the lazy match in `to:.*?\)` is necessary? wouldn't `to:.*\)` match to the last `)`?

**Review by joshuali925 — APPROVED** [review]
left question regarding regex

File: kibana-reports/public/components/main/report_details/report_details.tsx (hunk 2 above)
Comment (zhongnansu) [reply kibana-reports/public/components/main/report_details/report_details.tsx:125]: image

**Review by zhongnansu — APPROVED** [review]
Address the comment before merge

File: kibana-reports/public/components/main/report_details/report_details.tsx (hunk 2 above)
Comment (davidcui-amzn) [reply kibana-reports/public/components/main/report_details/report_details.tsx:125]: Good catch - will make change

**Pull request merged**